### PR TITLE
Zadanie 2

### PR DESCRIPTION
**Pull request by Algiz25**
Wykonałem zadanie 2. Wrzuciłem je wcześniej zanim pr'y były pozamykane, dlatego historia taka dziwna. Sorki za to. Ważny jest jedynie ostatni commit.

### CODE REVIEW

**Comment by shang1410** [comment]
Skalowanie danych ok, można ewentualnie przez zeskalowaniem Fare ją zlogarytmizować.
Brak wykresków, poza tym reszta ok.
Na plus użycie optuny do optymalizacji hiperparametrów.
Sprawdzenie szans na Titanicu - fajny dodatek ;)

**Review by dwartalski — APPROVED** [review]
podoba mi się:
	-zadbanie rzeczy tj. aby nie było data leaku, shuffle w train_test_splicie
	-użycie więcej niż 2 modeli
	-cross validacja
	-optymalizacja hipeparametrów 
	-danie własnej szansy na tytanikiu
Drobne uwagi:
	 -niezlogarytmowanie fare

**Review by MaciekPie — APPROVED** [review]
Fajnie zrobione. Podoba mi się użycie Optuny razem z Pipeline, gdzie skalowanie liczone jest osobno w każdym foldzie, i ładne wykresy (optuna.visualization). W train_test_split brakuje random_state.

**Comment by slovik02** [comment]
Na początku chcę zaznaczyć że nie ma przecieku danych, good job (tylko pamiętaj na przyszłość o tym random state w splicie, rozwiąże ci to trochę problemów)! Dodatkowe plusy za użycie optuny i zrobienie dodatkowego modelu, fajnie że wychodzisz z inicjatywą poza bare minimum ; )
Tylko trochę szkoda że nie ma wyjaśnienia czym się kierował*ś przy wyborze tych modeli.

Zwróć uwagę że w cross validation nie używasz przeskalowanych danych. Trochę też mi ogólnikowo brakuje komentarzy dlaczego coś robisz (na jakiej podstawie została podjęta twoja decyzja) oraz ogólnie co otrzymujemy z wyników i dlaczego
